bin/ch_misc.c

```diff
@@ -579,6 +636,17 @@ char *realpath_safe(const char *path)
    return pathc;
 }
 
+/* Replace all instances of character 'old' in array 'str' with 'new'. */
```

Comment:
Also document memory allocation behavior. As written, this implies that we modify `str`.